projects/angular-grid-layout/src/lib/grid.component.ts

```diff
@@ -200,7 +218,7 @@ export class KtdGridComponent implements OnChanges, AfterContentInit, AfterConte
          * binded in the component with it's input [layout]. In this scenario, we would always calculate one unnecessary change on the layout when
          * it is re-binded on the input.
          */
-        this._layout = layout;
+        this._layout = getUpdatedDimentions(layout, this.cols);
```

Comment:
Whether or not check for layout validity, in this case for min and max sizes, is out of the scope of this issue. IMO it is not clear for now that layout validation needs to be done by the grid, it may, but is not clear. Also, the vast majority of times this function will be executed, it would be of no use, because the user is just setting a valid layout with items with correct sizes. So, if this functionality gets introduced in the future, it may be flagged.

So, if you can remove it, it would be nice. I may open an issue for discussing this use case. thanks!